src/parser/mod.rs

```diff
@@ -617,6 +623,9 @@ impl<'a> Parser<'a> {
                 }
                 // `COMMENT` is snowflake specific https://docs.snowflake.com/en/sql-reference/sql/comment
                 Keyword::COMMENT if self.dialect.supports_comment_on() => self.parse_comment(),
+                Keyword::GO if dialect_of!(self is MsSqlDialect) => {
```

Comment:
```suggestion
                Keyword::GO => {
```
I think it should be fine to let the parser always accept the statement if it shows up. Technically if we wanted to gate it behind a feature then we could use a `self.dialect.supports` flag since the `dialect_of` macro is deprecated, but I think it makes more sense to let the parser accept unconditionally

Reply:
Done 👍